### ISSUE TITLE
[BUG] folder creation always lowers the foldername

### ISSUE THREAD

**Issue by Mik3yZ** [Checkmk/ansible-collection-checkmk.general#569]
**Describe the bug**
When creating a folder with uppercase letters, the folder.py changes it to lowercase.

**Component Name**
Component Name: folder

**Ansible Version**
```console
$ ansible --version
ansible [core 2.14.13]
  config file = /home/user/repos/ansible-checkmk/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/user/venvs/ansible-checkmk/lib64/python3.9/site-packages/ansible
  ansible collection location = /home/user/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/user/venvs/ansible-checkmk/bin/ansible
  python version = 3.9.16 (main, May 29 2023, 00:00:00) [GCC 11.3.1 20221121 (Red Hat 11.3.1-4)] (/home/user/venvs/ansible-checkmk/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```

**Checkmk Version and Edition**
```console
Checkmk Enterprise Edition 2.2.0p20
```

**Collection Version**

```console
$ ansible-galaxy collection list | grep checkmk.general
checkmk.general        4.3.1  
```

**To Reproduce**
Create a folder with:
``` yml
- name: "Create folder"
  checkmk.general.folder:
    server_url: "{{ master_site_baseurl }}"
    site: "{{ master_site }}"
    automation_user: "{{ automation_user }}"
    automation_secret: "{{ automation_secret }}"
    path: "/foldername"
    name: "FolderName"
    state: "present"
```

**Expected behavior**
A folder named FolderName, with path /foldername 

**Actual behavior**
A folder named foldername with path /foldername

**Minimum reproduction example**
See above

**Additional context**
`folder.py` contains:

https://github.com/Checkmk/ansible-collection-checkmk.general/blob/300764eaccd4b744908eb5415b9735a86d540d57/plugins/modules/folder.py#L220-L221

And this function (`_normalize_path`) changes the string to lowercase:

https://github.com/Checkmk/ansible-collection-checkmk.general/blob/300764eaccd4b744908eb5415b9735a86d540d57/plugins/modules/folder.py#L274-L278

I would advise to not normalize the name / title as a path.

**Comment by msekania** [comment]
Hi @Mik3yZ,

I would say works as intended.

Folder title and path (folder name) are two different parameters.
`path` (folder name)  is and to best of my knowledge should be low case.
`title` (alias `name`) on the other hand, can also contain capital letters.

**Comment by Mik3yZ** [comment]
So, to come back to your last sentence @msekania, `title` (alias `name` in the module) is parsed through function `_normalize_path`, and thus alway transformed to lowercase.

**Comment by msekania** [comment]
> So, to come back to your last sentence @msekania, `title` (alias `name` in the module) is parsed through function `_normalize_path`, and thus alway transformed to lowercase.

lines 220-221:  _normalize_path(...) is called with a parameter `path`, hence `path` is parsed with _normalize_path(...) not the name (alias title).

**Comment by Mik3yZ** [comment]
true, however it does set `(self.desired["parent"], self.desired["name"])` with that processed value. please run my example with 4.3.1 and see for yourself.

**Comment by msekania** [comment]
https://github.com/Checkmk/ansible-collection-checkmk.general/blob/300764eaccd4b744908eb5415b9735a86d540d57/plugins/modules/folder.py#L223C1-L224C1

states it explicitly what goes to `title`

**Comment by Mik3yZ** [comment]
yes, i saw that too. however i do get only lowercase names while creating the folders.

i would then say line 220 needs to be changed from:

`(self.desired["parent"], self.desired["name"]) = self._normalize_path(`
to
`(self.desired["parent"], self.desired["path"]) = self._normalize_path(`

**Comment by msekania** [comment]
> yes, i saw that too. however i do get only lowercase names while creating the folders.
> 
> i would then say line 220 needs to be changed from:
> 
> `(self.desired["parent"], self.desired["name"]) = self._normalize_path(` to `(self.desired["parent"], self.desired["path"]) = self._normalize_path(`

`self.desired["title"] = self.params.get("title", self.desired["name"]) `
No, actually that's fine, because if title is not defined `self.desired["name"]` folder is default name

problem is somewhere else, while `self.params.get("title")` is `Null` instead of Name.

**Comment by msekania** [comment]
It should be
`self.desired["title"] = self.params.get("name", self.desired["name"])`

I'll make corrections

**Comment by msekania** [comment]
@Mik3yZ ,

see the fix #570 

Thanks for reporting the issue!
